lib/plugins/aws/package/compile/events/websockets/index.js

```diff
@@ -12,6 +12,9 @@ const compileDeployment = require('./lib/deployment');
 const compileStage = require('./lib/stage');
 const compileAuthorizers = require('./lib/authorizers');
 
+const standardRoutes = new Set(['connect', 'disconnect', 'default']);
+const routePattern = new RegExp(`^(?:[^$].*|\\$(${Array.from(standardRoutes).join('|')}))$`, 'i');
```

Comment:
I haven't been able to find an official AWS source on this but whenever I try to create a custom websocket route starting with $, I get an error from ApiGateway saying the route is not formatted properly. 

E.g., if I try to create a route called "$foo" in SLS, I get the following error: 

An error occurred: SfooWebsocketsRoute - The provided route key is not formatted properly (Service: AmazonApiGatewayV2; Status Code: 400; Error Code: BadRequestException; Request ID: ...; Proxy: null)

Thinking about it, it's probably better to remove the pattern validation from the schema since there's no clear source on how AWS validates route names and API Gateway already returns an error that explains the problem anyways so I removed it!

Reply:
> Thinking about it, it's probably better to remove the pattern validation from the schema since there's no clear source on how AWS validates route names 

Yes, definitely. I believe we should imply a full validation only if we have a straightforward reference on AWS side, otherwise let AWS do validation in their scope